libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity/Scripts/PeerConnection.cs

```diff
@@ -1,608 +1,1208 @@
+        {
+            return string.Format("{0}:{1}", Type.ToString().ToLowerInvariant(), Uri);
+        }
+    }
+
+    /// <summary>
+    /// A <a href="https://docs.unity3d.com/ScriptReference/Events.UnityEvent.html">UnityEvent</a> that represents a WebRTC error event.
+    /// </summary>
+    [Serializable]
+    public class WebRTCErrorEvent : UnityEvent<string>
+    {
+    }
+
+    /// <summary>
+    /// Media line abstraction for a peer connection.
+    /// 
+    /// This container binds together a sender component (<see cref="MediaSender"/>) and/or a receiver component
+    /// (<see cref="MediaReceiver"/>) to a media line, and therefore once a negotiation was completed to the
+    /// transceiver attached to that media line.
+    /// 
+    /// Assigning Unity components to the <see cref="Sender"/> and <see cref="Receiver"/> fields serves
+    /// as an indication of the user intent to send and/or receive media through the transceiver, and is
+    /// used during negotiation to derive the <see xref="WebRTC.Transceiver.Direction"/> to negotiate.
+    /// After the negotiation is completed, the <see cref="Transceiver"/> property refers to the transceiver
+    /// associated with this media line, and which the sender and receiver will use.
+    /// </summary>
+    [Serializable]
+    public class MediaLine
+    {
+        /// <summary>
+        /// Backing field for the <see cref="Kind"/> property.
+        /// </summary>
+        /// <seealso cref="Kind"/>
+        [SerializeField]
+        private MediaKind _kind;
+
+        /// <summary>
+        /// Kind of media of the media line and its attached transceiver.
+        /// This is assiged when the media line is created with <see cref="PeerConnection.AddTransceiver(MediaKind)"/>
+        /// and is immutable for the lifetime of the peer connection.
+        /// </summary>
+        public MediaKind Kind => _kind;
+
+        /// <summary>
+        /// Media sender producing the media to send through the transceiver attached to this media line.
+        /// This must be an instance of a class derived from <see cref="AudioSender"/> or <see cref="VideoSender"/>
+        /// depending on <see cref="Kind"/>.
+        /// If this is non-<c>null</c> then the peer connection will negotiate sending some media, otherwise
+        /// it will signal the remote peer that it does not wish to send (receive-only or inactive).
+        /// </summary>
+        public MediaSender Sender;
+
+        /// <summary>
+        /// Media receiver consuming the media received through the transceiver attached to this media line.
+        /// This must be an instance of a class derived from <see cref="AudioReceiver"/> or <see cref="VideoReceiver"/>
+        /// depending on <see cref="Kind"/>.
+        /// If this is non-<c>null</c> then the peer connection will negotiate receiving some media, otherwise
+        /// it will signal the remote peer that it does not wish to receive (send-only or inactive).
```

Comment:
nit: it is a bit unclear what happens if this is changed after the negotiation has happened.

Reply:
Good catch. Currently this requires a renegotiation, and the `RenegotiationNeeded` event is **not** raised. This is a bit unexpected I think. Maybe changing those values should immediately change the transceiver's desired direction for consistency, which would raise the event. However this only works after the first negotiation, because before it the transceiver is not yet available, so this introduces some inconsistency. Thoughts?

Reply:
Discussed in chat, changing these members should change the direction on the corresponding transceiver. It's fine if it does nothing before the first negotiation, this (and the fact that the native objects should be used with care) will be documented.

Reply:
Fixed.